src/algorithm/dyadic/mod.rs

```diff
@@ -1785,3 +1785,59 @@ impl<T: ArrayValue> Array<T> {
         Ok(Array::new(shape, data))
     }
 }
+
+impl Value {
+    /// Test if a value is an integer in a range
+    pub fn memberof_range(&self, from: Self, env: &Uiua) -> UiuaResult<Self> {
+        fn fallback(a: &Value, b: &Value, env: &Uiua) -> UiuaResult<Value> {
+            let range = a.range(env)?;
+            b.member(&range, env)
+        }
+
+        let Ok(range_bound) = self.as_num(env, "") else {
```

Comment:
I think you should use `.as_nat` here, which returns a `usize` and would eliminate a lot of the checks you do later.

Reply:
I chose to do it this way because I wanted to optimize negatives as well

Reply:
Negatives were included in my original use case. I did have `as_nat` before I got to including them

Reply:
Ah, okay that makes sense